02-algorithms/2-Searching/algos_searching_answers.txt

```diff
@@ -0,0 +1,22 @@
+1. Using linear search it would take 7 iterations to find G. 
+2. Using Binary search it would take 3 iterations to find G.
+3. (I’m confused why we’re doing this or how to demonstrate that I’m really doing this by hand but) the fib sequence is (I’m adding them in my head you’ll just have to believe me I guess (or maybe I’ll have screwed one up?) ??? 
+	1) 1 
+	2) 1 
+	3) 2 
+	4) 3
+	5) 5 
+	6) 8 
+	7) 13
+	8) 21
+	9) 34
+	10) 55
+	11) 89
+	12) 144
```

Comment:
The point of this is to actually understand the process of the Fibonacci sequence as far as how you do it. This process often helps lead to better solutions, as we tend to over-complicate things if we run with pure code. For example, you realize that you only need to know the last two sums to get the next sum, so why are we recalculating the whole thing with each previous sum when we implement Fib in code? Our recursive solution, while simpler _looking_ in code, is vastly more complicated if you tried to write out the recursion by hand. The iterative solution follows much more closely with the human process, and ends up being a lot simpler complexity-wise (and more efficient, of course).